instruqt-tracks/nomad-integration-with-vault/track.yml

```diff
@@ -0,0 +1,490 @@
+    type: service
+    hostname: hashistack-server
+    port: 8500
+  - title: Vault
+    type: service
+    hostname: hashistack-server
+    port: 8200
+  difficulty: basic
+  timelimit: 300
+- slug: deploy-an-application
+  id: 4ddvvt3xlnzr
+  type: challenge
+  title: Deploy an Application
+  teaser: Deploy an Application with the Appropriate Policy and Templating
+  assignment: |-
+    The Nomad job file has been also been created for you and you can take a look at it by clicking on the <b>"Files"</b> tab and navigating to <b>hashistack > vault > web-app.nomad</b>.</ br></ br>
+    Run the Nomad job by running the command;
+    ```
+    nomad run /root/hashistack/nomad/web.nomad
+    ```
+    Click on the <b>"Nomad"</b> tab and wait until the job has been deployed and healthy.</ br></ br>
+    Confirm that the application is accessing the database by running two commands. Use the dig command to query the SRV record of your service and obtain the port it is using;
+    ```
+    dig +short SRV web.service.consul.
+    ```
+    The response should look similar to <b>1 1 30478 ip-172-31-58-230.node.instruct.consul.</b>. Next curl your service at the appropriate port and names path;
+    ```
+    curl http://web.service.consul:3000/api
+    ```
+    The response should return json from the api.<br /><br />
+    Finally, click of the <b>"Application"</b> tab to view the web application.<br /><br />
+    Click the <b>"Check"</b> button in the lower right to complete the challenge.
+  notes:
+  - type: text
+    contents: |-
+      <h1>Deploy an Application</h1>
+      <hr />
+      <p>Now we are ready to deploy our web application and give it the necessary policy and configuration to communicate with our database.</p>
+  - type: text
+    contents: |-
+      <p>There are a few key points to take a look at in the <b>/root/hashistack/nomad/web.nomad</b> file;</p>
+      <p>We have specified the access-tables policy in the vault stanza of this job. The Nomad client will receive a token with this policy attached. Recall from the previous step that this policy will allow our application to read from the database/creds/accessdb endpoint in Vault and retrieve credentials.</p>
+      <pre style="overflow-x: auto; white-space: pre-wrap; white-space: -moz-pre-wrap; white-space: -pre-wrap; white-space: -o-pre-wrap; word-wrap: break-word; padding: 15px;">
+      vault {
+        policies = ["access-tables"]
+      }
+      </pre>
+  - type: text
+    contents: <p>We are using the template stanza's vault integration to populate
+      environment variables file that our application needs. The underlying tool being
+      used is Consul Template. You can use Consul Template's documentation to learn
+      more about the syntax needed to interact with Vault. Please note that although
+      we have defined the template inline, we can use the template stanza in conjunction
+      with the artifact stanza to download an input template from a remote source
```

Comment:
I'm not sure you really need to mention the possibility of downloading templates with the artifact stanza.